### PR TITLE
[JP Social] Update UI for Post Settings screen

### LINKED ISSUE #18571: [JP Social/Post Settings] JP Social connections and message
### Tasks
- [x] Create a "Jetpack Social" section inside the post settings page https://github.com/wordpress-mobile/WordPress-Android/pull/18708
- [x] Display list of connected accounts
- [x] Show share message and let the user customize it

_Note: the toggle implementation is in a different task, as it might require changes in API requests._

### PR DESCRIPTION
**Pull request by RenanLukas**
Part of #18571 

To test:
Nothing to test yet, except the UI. Please run `EditPostSettingsJetpackSocialContainerConnectionsListViewPreview` and `EditPostSettingsJetpackSocialNoConnectionsPreview`.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
--

3. What automated tests I added (or what prevented me from doing so)
--

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    postSocialConnectionList: List<PostSocialConnection>,
+    shareMessage: String,
+    remainingSharesMessage: String,
+    subscribeButtonLabel: String,
+    onSubscribeClick: () -> Unit,
+    connectProfilesButtonLabel: String,
+    onConnectProfilesCLick: () -> Unit,
+) {
+    Column {
+        if (postSocialConnectionList.isNotEmpty()) {
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt]: This decision will be moved to the ViewModel in another PR.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialConnectionsList.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    Column(
+        Modifier
+            .fillMaxWidth()
+    ) {
+        postSocialConnectionList.forEachIndexed { _, connection ->
+            PostSocialConnectionItem(
+                connection = connection,
+                onSharingChange = {},//TODO
+            )
+            Divider()
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialConnectionsList.kt:27]: 💡 Do we want this trailing `Divider` for all items, including the last one? In case we don't, we can use the `index` to avoid showing for the last one, wdyt?

In case we want for all items then we might want to use `forEach` instead of `forEachIndexed`.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                    connectProfilesButtonLabel = it.connectProfilesButtonLabel,
+                    onConnectProfilesCLick = it.onConnectProfilesCLick,
+                )
+            }
+        }
+    }
+
+    fun update(data: EditPostSettingsJetpackSocialContainerComposeViewData) {
+        this.data = data
+        disposeComposition()
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt]: 💡 Suggestion: instead of doing this imperatively by disposing the composition I think it would be better to store the `data` in a `MutableState` so `update()` can update that state and that would already cause a recomposition. [Here's an example](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/compose/views/TrainOfAvatarsView.kt) of a Compose View Wrapper doing that. Does that make sense?

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import org.wordpress.android.ui.compose.components.buttons.PrimaryButton
+import org.wordpress.android.ui.compose.theme.AppThemeEditor
+import org.wordpress.android.ui.compose.unit.Margin
+import org.wordpress.android.ui.posts.social.PostSocialConnection
+import org.wordpress.android.ui.posts.social.compose.PostSocialMessageItem
+import org.wordpress.android.ui.posts.social.compose.PostSocialSharesText
+import org.wordpress.android.ui.publicize.PublicizeServiceIcon
+
+@Composable
+fun EditPostSettingsJetpackSocialContainerView(
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt]: ⚠️  I would avoid using the suffix `View` for Composables as it might be confused with View-based UIs/components, which usually have that suffix (e.g.: `TextView`, `ImageView`, etc). Also, since we are using both XMLs (view-based) and Compose in this codebase I would reserve the View suffix for View-wrappers around Composables if needed.

In this case, the file names are really confusing because the `EditPostSettingsJetpackSocialContainerComposeView` has `Compose` in the name but it is NOT a Composable, while this houses a Composable but that has `View` in the name.

I suggest renaming this component to `EditPostSettingsJetpackSocialContainer`, and renaming the view wrapper to `EditPostSettingsJetpackSocialContainerView`.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    trainOfIconsModels: List<TrainOfIconsModel>,
+    postSocialConnectionList: List<PostSocialConnection>,
+    shareMessage: String,
+    remainingSharesMessage: String,
+    subscribeButtonLabel: String,
+    onSubscribeClick: () -> Unit,
+    connectProfilesButtonLabel: String,
+    onConnectProfilesCLick: () -> Unit,
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt]: 💡 Maybe it's worth receiving `modifier` as well, especially since callers of this Composable might want it to fill the entire width.

**Review by thomashorta — CHANGES_REQUESTED** [review]
Thanks for the PR, Renan!
I added some minor comments and a concern that might require changes.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialConnectionsList.kt (hunk 2 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialConnectionsList.kt:27]: That's a good point. Initially I wasn't showing the last line using the index, like you said, but then I realized the design specs show the divider even for the last row. I believe the reason might be to have a clear separation between the last row and the `Message` container. WDYT?

image

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt (hunk 3 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt]: Good idea 👍  if it's OK for you, I'll handle this in my next PR that implements the VM changes and starts using this View.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt (hunk 4 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt]: Good catch, thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt (hunk 5 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerView.kt]: Makes sense 👍 thanks

**Comment by RenanLukas** [comment]
Thanks for the review, @thomashorta . I've updated the code and replied to the comments. Please let me know if it makes sense 👍

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt (hunk 3 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt]: Sounds good!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainer.kt [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import org.wordpress.android.ui.compose.theme.AppThemeEditor
+import org.wordpress.android.ui.compose.unit.Margin
+import org.wordpress.android.ui.posts.social.PostSocialConnection
+import org.wordpress.android.ui.posts.social.compose.PostSocialMessageItem
+import org.wordpress.android.ui.posts.social.compose.PostSocialSharesText
+import org.wordpress.android.ui.publicize.PublicizeServiceIcon
+
+@Composable
+fun EditPostSettingsJetpackSocialContainer(
+    modifier: Modifier = Modifier,
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainer.kt]: ⚠️ The order of the arguments seems swapped, `modifier` should be the first **optional** parameter in the Composable function, as noted in the [documentation](https://developer.android.com/reference/kotlin/androidx/compose/ui/Modifier#:~:text=It%20should%20appear%20as%20the%20first%20optional%20parameter%20in%20the%20parameter%20list%3B%20after%20all%20required%20parameters%20(except%20for%20trailing%20lambda%20parameters)%20but%20before%20any%20other%20parameters%20with%20default%20values.):
> It should appear as the first optional parameter in the parameter list; after all required parameters (except for trailing lambda parameters) but before any other parameters with default values.

So in this case it should be the last parameter in the function since it's the only optional parameter and the lambda parameters are not really meant to be trailing lambda parameters.

**Review by thomashorta — COMMENTED** [review]
Hey, thanks for renaming the Compose and View files! 🙇 

I just added a comment about the modifier parameter which should be quick to fix.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainer.kt (hunk 6 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainer.kt]: Good point. Updated 👍

**Review by thomashorta — APPROVED** [review]
LGTM! Nice job!!! :shipit:

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt (hunk 3 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialContainerComposeView.kt]: [Done](https://github.com/wordpress-mobile/WordPress-Android/pull/18781/files#diff-3c86b940c933439b5e96091155d817bd64b0a658d5d8e477920ad3d66813d0cbR19)
